### ISSUE TITLE
[Bug] Track Humanoids (BB #713)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#713]
This issue was migrated from bitbucket.
**Original Reporter:** Flywithme
**Original Date:** 11.10.2014 07:40:02 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713

<hr>

http://tbc.wowroad.info/?spell=5225 - Not working properly.

How it should work: auto casts when you enter Cat Form, unless you're already tracking something else.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 11.10.2014 11:18:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-12906556

<hr>

Any proof it should work like that?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 11.10.2014 11:33:52 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-12906676

<hr>

Comment Patch 2.0.12: http://www.wowhead.com/spell=5225/track-humanoids#comments:id=63446

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 11.10.2014 16:01:39 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-12908775

<hr>

I can confirm.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 11.01.2015 14:20:04 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-14736162

<hr>

SpellAuras.cpp

```
#!c++
m_target->SetPower(POWER_RAGE,0);
if (urand(1,100) <= FurorChance)
m_target->CastSpell(m_target, 17057, true, NULL, this);
+ }
+ 
+ if (form == FORM_CAT)
+ {
+ if(m_target->HasSpell(5225) && !m_target->GetUInt32Value(PLAYER_TRACK_CREATURES) && !m_target->GetUInt32Value(PLAYER_TRACK_RESOURCES))
+ m_target->CastSpell(m_target, 5225, true);
}
break;
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 11.01.2015 14:46:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-14736454

<hr>

I guess it should untrack humanoids when you leave the form?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 11.01.2015 14:51:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-14736532

<hr>

Yes, track humanoids when you enter Cat Form, and untrack humanoids when you leave the form

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 11.01.2015 16:20:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/713/#comment-14737489

<hr>

Fix #713 - thanks to Flywithme

→ <<cset 6b652031dd79>>
